### PR TITLE
Update harmony remote for new aioharmony library

### PR DESCRIPTION
**Pull request by ehendrix23**
**Description:**

-) Updated example as configuration file written is now in JSON format.
-) Added that Harmony configuration file is now automatically updated anytime the Harmony HUB configuration is updated.
-) Improve send_command example as it now also allows for the name of the device and not just device ID.

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#19595

## Checklist:

- [x] Branch: `next` is for changes and new documentation that will go public with the next [home-assistant](https://github.com/home-assistant/home-assistant) release. Fixes, changes and adjustments for the current release should be created against `current`.
- [x] The documentation follows the [standards][standards].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

File: source/_components/remote.harmony.markdown [hunk 1]
```diff
@@ -97,8 +99,14 @@ Start an activity. Will start the default `activity` from configuration.yaml if
 In the file 'harmony_REMOTENAME.conf' you can find the available activities, for example:
 
 ```text
-Activities
-  22054907 - Watch TV
+{
+    "Activities": {
```
Comment (MartinHjelmare) [inline source/_components/remote.harmony.markdown:103]: Were the docs stale from before this change? The merged PR didn't change the config file format, right?

Comment (ehendrix23) [reply source/_components/remote.harmony.markdown:103]: The config file that is written is in JSON format now. 
The file itself is not used by HASS but is rather just informative so people can reference to write services etc. to know when to write services etc. what different devices and commands are supported.


That was in the PR for the Harmony fix

Comment (MartinHjelmare) [reply source/_components/remote.harmony.markdown:103]: But it already saved the file as json? If we did change this, it's a breaking change, so it's important that we are clear about this.

Comment (ehendrix23) [reply source/_components/remote.harmony.markdown:103]: It did as from PR #19440 as at that time it was faster for me to get the websocket fixes in based on how pyharmony was done and other working in-progress code I had.

Comment (MartinHjelmare) [reply source/_components/remote.harmony.markdown:103]: Ah ok, so this is just correcting the docs from the earlier PR change.

Is this file meant for manual editing too?

Comment (ehendrix23) [reply source/_components/remote.harmony.markdown:103]: Correct. 

No, this file is informational so people know what type of commands can be sent etc.
The file is overwritten anytime HASS starts or configuration change on the Harmony itself.

**Review by MartinHjelmare — APPROVED** [review]

**Pull request merged**